RDPの画面サイズ設定が未設定の場合、例外が発生する

RDPの画面サイズ設定が未設定の場合、接続時に例外が発生します。
この現象は、 v1.0.6 以降で発生するようです。

ご確認、ご対応のほど、よろしくお願いします。

### 設定内容

![2020-03-23_15h09_09](https://user-images.githubusercontent.com/25142620/77288029-4c020700-6d1a-11ea-852a-b61ea91a29e5.png)

### エラーダイアログ

![2020-03-23_15h09_19](https://user-images.githubusercontent.com/25142620/77288076-5fad6d80-6d1a-11ea-9e89-6d7f2038cc57.png)

### エラーログ

```
2020-03-23 15:04:27.4292 [ERROR] Failed to connect to name (192.0.2.1:3389). - kenzauros.RHarbor.ViewModels.ConnectionInfoManagementViewModel`1+<Connect>d__57.MoveNext() System.InvalidOperationException: Null 許容のオブジェクトには値を指定しなければなりません。
   場所 System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   場所 kenzauros.RHarbor.Models.RDPConnectionInfo.SaveAs(String filepath, String host, Nullable`1 port)
   場所 kenzauros.RHarbor.RDPClient.Open(String actualHost, Nullable`1 actualPort)
   場所 kenzauros.RHarbor.ViewModels.RDPConnectionViewModel.<Connect>d__6.MoveNext()
--- 直前に例外がスローされた場所からのスタック トレースの終わり ---
   場所 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   場所 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   場所 kenzauros.RHarbor.ViewModels.ConnectionInfoManagementViewModel`1.<Connect>d__57.MoveNext()
```

--- a/RHarbor/Models/RDPConnectionInfo.cs
+++ b/RHarbor/Models/RDPConnectionInfo.cs
@@ -91,10 +91,15 @@ namespace kenzauros.RHarbor.Models
         /// <param name="port"></param>
         public void SaveAs(string filepath, string host = null, int? port = null)
         {
-            var workArea = System.Windows.SystemParameters.WorkArea;
-            var (w, h) = (DesktopWidth.Value + 50, DesktopHeight.Value + 100);
-            var (l, t) = (Math.Floor((workArea.Width - w) / 2), Math.Floor((workArea.Height - h) / 2));
-            var (r, b) = (l + w, t + h);
+            var winposstr = string.Empty;
+            if (DesktopWidth.HasValue && DesktopHeight.HasValue)
+            {
+                var workArea = System.Windows.SystemParameters.WorkArea;
+                var (w, h) = (DesktopWidth.Value + 50, DesktopHeight.Value + 100);
+                var (l, t) = (Math.Floor((workArea.Width - w) / 2), Math.Floor((workArea.Height - h) / 2));
+                var (r, b) = (l + w, t + h);
+                winposstr = $"winposstr:s:0,1,{l},{t},{r},{b}";
+            }
             File.WriteAllText(filepath,
                 $@"
 screen mode id:i:{(FullScreen ? 2 : 1)}
@@ -103,7 +108,7 @@ screen mode id:i:{(FullScreen ? 2 : 1)}
 username:s:{Username}
 use multimon:i:0
 session bpp:i:32
-winposstr:s:0,1,{l},{t},{r},{b}
+{winposstr}
 compression:i:1
 keyboardhook:i:2
 audiocapturemode:i:0

--- a/RHarbor/Properties/AssemblyInfo.cs
+++ b/RHarbor/Properties/AssemblyInfo.cs
@@ -51,4 +51,4 @@ using System.Windows;
 // すべての値を指定するか、次を使用してビルド番号とリビジョン番号を既定に設定できます
 // 既定値にすることができます:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("1.1.0")]
+[assembly: AssemblyVersion("1.1.1")]

--- a/RHarbor/Properties/Resources.Designer.cs
+++ b/RHarbor/Properties/Resources.Designer.cs
@@ -569,7 +569,7 @@ namespace kenzauros.RHarbor.Properties {
         }
         
         /// <summary>
-        ///   Private key file does not exist. に類似しているローカライズされた文字列を検索します。
+        ///   Private key file does not exist. &quot;{0}&quot; に類似しているローカライズされた文字列を検索します。
         /// </summary>
         public static string SSHConnection_Exception_PrivateKeyFileNotExists {
             get {

--- a/RHarbor/Properties/Resources.ja-JP.resx
+++ b/RHarbor/Properties/Resources.ja-JP.resx
@@ -384,7 +384,7 @@
     <value>ポート番号は 1 から 65535 の範囲でなければなりません。</value>
   </data>
   <data name="SSHConnection_Exception_PrivateKeyFileNotExists" xml:space="preserve">
-    <value>鍵ファイルが見つかりません。</value>
+    <value>鍵ファイルが見つかりません。 "{0}"</value>
   </data>
   <data name="SSHConnection_Exception_RemoteHostPortNotAssigned" xml:space="preserve">
     <value>リモートホストとポートが指定されていません。</value>

--- a/RHarbor/Properties/Resources.resx
+++ b/RHarbor/Properties/Resources.resx
@@ -375,7 +375,7 @@ This operation cannot be canceled.</value>
     <value>Port number should be between 1 and 65535.</value>
   </data>
   <data name="SSHConnection_Exception_PrivateKeyFileNotExists" xml:space="preserve">
-    <value>Private key file does not exist.</value>
+    <value>Private key file does not exist. "{0}"</value>
   </data>
   <data name="SSHConnection_Exception_RemoteHostPortNotAssigned" xml:space="preserve">
     <value>Remote Host and Port should be assigned.</value>

--- a/RHarbor/ViewModels/ConnectionInfoManagementViewModels/ConnectionInfoManagementViewModel.cs
+++ b/RHarbor/ViewModels/ConnectionInfoManagementViewModels/ConnectionInfoManagementViewModel.cs
@@ -230,15 +230,19 @@ namespace kenzauros.RHarbor.ViewModels
         protected virtual async Task Connect(T item)
         {
             MyLogger.Log($"Connecting to {item.ToString()}...");
-            var conn = ConnectionViewModel<T>.CreateFromConnectionInfo(item);
-            MainWindow.Connections.Collection.Add(conn);
+            ConnectionViewModel<T> conn = null;
             try
             {
+                conn = ConnectionViewModel<T>.CreateFromConnectionInfo(item);
+                MainWindow.Connections.Collection.Add(conn);
                 await conn.Connect();
             }
             catch (Exception ex)
             {
-                MainWindow.Connections.Collection.Remove(conn);
+                if (conn != null)
+                {
+                    MainWindow.Connections.Collection.Remove(conn);
+                }
                 switch (ex)
                 {
                     case OperationCanceledException oce: // User canceled manually
@@ -247,7 +251,7 @@ namespace kenzauros.RHarbor.ViewModels
                 }
                 MyLogger.Log($"Failed to connect to {item.ToString()}.", ex);
                 await MainWindow.ShowMessageDialog(
-                    string.Format(Resources.ConnectionInfo_Dialog_Save_Error, item.ToString(), ex.Message),
+                    string.Format(Resources.ConnectionInfo_Dialog_Connect_Error, item.ToString(), ex.Message),
                     Resources.ConnectionInfo_Dialog_Connect_Title);
             }
         }

--- a/RHarbor/ViewModels/ConnectionViewModels/SSHConnectionViewModel.cs
+++ b/RHarbor/ViewModels/ConnectionViewModels/SSHConnectionViewModel.cs
@@ -41,7 +41,9 @@ namespace kenzauros.RHarbor.ViewModels
             }
             if (!string.IsNullOrWhiteSpace(info.PrivateKeyFilePath) && !File.Exists(info.PrivateKeyFilePath))
             {
-                throw new FileNotFoundException(Resources.SSHConnection_Exception_PrivateKeyFileNotExists, info.PrivateKeyFilePath);
+                throw new FileNotFoundException(
+                    string.Format(Resources.SSHConnection_Exception_PrivateKeyFileNotExists, info.PrivateKeyFilePath),
+                    info.PrivateKeyFilePath);
             }
             if (createRegisteredForwardedPorts && info.PortForwardingCollection != null)
             {

--- a/packaging.ps1
+++ b/packaging.ps1
@@ -5,7 +5,7 @@ $releaseDir = Join-Path $directory "Release" -Resolve
 $filename = $appName + ".exe"
 $exePath = Join-Path $releaseDir $filename -Resolve
 $fileVersionInfo = (Get-ItemProperty $exePath).VersionInfo
-$fileVersion = "v" + ($fileVersionInfo.FileMajorPart) + "." + ($fileVersionInfo.FileMinorPart) + "." + ($fileVersionInfo.FilePrivatePart)
+$fileVersion = "v" + ($fileVersionInfo.FileMajorPart) + "." + ($fileVersionInfo.FileMinorPart) + "." + ($fileVersionInfo.FileBuildPart)
 $destDir = Join-Path $directory ($appName + "_" + $fileVersion)
 $destPath = $destDir + ".zip"
 

